addons/website_sale_delivery/models/sale_order.py

```diff
@@ -1,121 +1,83 @@
-            store={
-                'sale.order': (lambda self, cr, uid, ids, c={}: ids, ['order_line'], 10),
-                'sale.order.line': (_get_order, ['price_unit', 'tax_id', 'discount', 'product_uom_qty'], 10),
-            },
-            help="Has an order line set for delivery"
-        ),
-        'website_order_line': fields.one2many(
-            'sale.order.line', 'order_id',
-            string='Order Lines displayed on Website', readonly=True,
-            domain=[('is_delivery', '=', False)],
-            help='Order Lines to be displayed on the website. They should not be used for computation purpose.',
-        ),
-    }
-
-    def _check_carrier_quotation(self, cr, uid, order, force_carrier_id=None, context=None):
-        carrier_obj = self.pool.get('delivery.carrier')
-
+    amount_delivery = fields.Monetary(
+        compute='_compute_amount_delivery', digits=0,
+        string='Delivery Amount',
+        help="The amount without tax.", store=True, track_visibility='always'
+    )
+    has_delivery = fields.Boolean(
+        compute='_compute_has_delivery', string='Has delivery',
+        help="Has an order line set for delivery", store=True
+    )
+    website_order_line = fields.One2many(
+        'sale.order.line', 'order_id',
+        string='Order Lines displayed on Website', readonly=True,
+        domain=[('is_delivery', '=', False)],
+        help='Order Lines to be displayed on the website. They should not be used for computation purpose.',
+    )
+
+    @api.depends('order_line', 'order_line.price_unit', 'order_line.tax_id', 'order_line.discount', 'order_line.product_uom_qty')
+    def _compute_amount_delivery(self):
+        for order in self:
+            order.amount_delivery = sum(order.order_line.filtered('is_delivery').mapped('price_subtotal'))
+
+    @api.depends('order_line', 'order_line.price_unit', 'order_line.tax_id', 'order_line.discount', 'order_line.product_uom_qty')
+    def _compute_has_delivery(self):
+        for order in self:
+            order.has_delivery = bool(order.order_line.filtered('is_delivery'))
+
+    def _check_carrier_quotation(self, force_carrier_id=None):
+        self.ensure_one()
         # check to add or remove carrier_id
-        if not order:
+        if not self:
             return False
-        if all(line.product_id.type in ("service", "digital") for line in order.website_order_line):
-            order.write({'carrier_id': None})
-            self.pool['sale.order']._delivery_unset(cr, SUPERUSER_ID, [order.id], context=context)
+        if self.only_services:
+            self.write({'carrier_id': None})
+            self._delivery_unset()
             return True
-        else: 
-            carrier_id = force_carrier_id or order.carrier_id.id
-            carrier_ids = self._get_delivery_methods(cr, uid, order, context=context)
+        else:
+            carrier_id = force_carrier_id or self.carrier_id.id
+            carrier_ids = self._get_delivery_methods()
```

Comment:
Because we are calling this method only twice to get ids. One of the call [here](https://github.com/odoo/odoo/blob/master/addons/website_sale_delivery/models/sale_order.py#L89) is to get list of ids and insertion/removal is [performed](https://github.com/odoo/odoo/blob/master/addons/website_sale_delivery/models/sale_order.py#L94) which can't be performed on recordset  and since code improvement is prohibited while migrating to new API, we will have to get the list of ids again if we return recordset here.